pkg/auth/oauth2authenticator.go

```diff
@@ -359,50 +403,40 @@ func (o *oAuth2Authenticator) authenticateWithAuthorizationCode() error {
 		// launch browser
 		go o.openBrowserFunc(authCodeUrl)
 
+		// Handle context cancellation
 		timedOut := false
-		timer := time.AfterFunc(TIMEOUT_SECONDS, func() {
-			timedOut = true
+		canceled := false
+		go func() {
+			<-ctx.Done()
+			switch err := ctx.Err(); {
+			case errors.Is(err, context.DeadlineExceeded):
+				timedOut = true
+			case errors.Is(err, context.Canceled):
+				canceled = true
+			}
```

Comment:
should this maybe just be executed after srv.Serve and not in a go func? I see a potential race for timeout/canceled.